test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -258,30 +262,120 @@ private static void getFailureReason() {
 
     /**
      * Position the instruction frame with testWindow (testcase created
-     * window) by the specified position.
+     * window) by the specified position. If testWindow is null, only
+     * the instruction frame is positioned according to the Position
+     * parameter.
      * Note: This method should be invoked from the method that creates
-     * testWindow.
+     * testWindow. At test-level, the testWindow must be made visible
+     * after calling this method.
+     *
+     * @param testWindow test window that the test is created. It can be null.
+     * @param position  position can either be:
+     *                  HORIZONTAL - the test instruction frame is positioned
+     *                  such that its right edge aligns with screen's vertical
+     *                  center and the test window (if not null) is placed to
+     *                  the right of the instruction frame.
+     *
+     *                  VERTICAL - the test instruction frame is positioned such
+     *                  that its bottom edge aligns with the screen's horizontal
+     *                  center and the test window (if not null) is placed below
+     *                  the instruction frame.
      *
-     * @param testWindow test window that the test is created
-     * @param position  position can be either HORIZONTAL (both test
-     *                  instruction frame and test window as arranged
-     *                  side by side) or VERTICAL (both test instruction
-     *                  frame and test window as arranged up and down)
+     *                  TOP_LEFT_CORNER - the test instruction frame is positioned
+     *                  such that its top left corner is at the top left corner of
+     *                  the screen and the test window (if not null) is placed to
+     *                  the right of the instruction frame.
      */
     public static void positionTestWindow(Window testWindow, Position position) {
         Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+
+        // to get the screen insets inorder to position the frame by taking into
+        // account the location of taskbar/menubars on screen
+        GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration();
+        Insets screenInsets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
+
```

Comment:
I suggest separating the logic to calculate coordinates into a separate function, or save them into a local variable.

Then call `sync` and `sleep`. Or this sequence could be extracted into its own helper method to avoid code duplication, it's repeated at least twice.

Shall this method call `window.setVisible(true)` automatically?

Reply:
@aivanov-jdk As suggested, separating the common code into a helper method would make the code look cleaner. 

Earlier I did think about having the setVisible for both the testWindow and instruction frame within the positionWindow but usually at the test-level, the user would be in the habit of creating the testWindow and setting its visibility to true when creating test UI. Hence left it to be set at test-level. I can change it, if this approach is better.

Either way we need to make minor changes to existing manual tests - **remove** (in case we add testWindow.setVisible(true) to PassFailJFrame) or **reposition** (in case we retain it at test-level).

Reply:
Toolkit.sync + Thread.sleep extracted to a separate helper method.

Reply:
> Earlier I did think about having the setVisible for both the testWindow and instruction frame within the positionWindow but usually at the test-level, the user would be in the habit of creating the testWindow and setting its visibility to true when creating test UI. Hence left it to be set at test-level. I can change it, if this approach is better.
> 
> Either way we need to make minor changes to existing manual tests - **remove** (in case we add testWindow.setVisible(true) to PassFailJFrame) or **reposition** (in case we retain it at test-level) setVisible() call.

@aivanov-jdk I have retained the `setVisible()` for testWindow on test side (where test UI is created) for reasons as described above. Please let me know in case it is better to have it within `positionWindow()`.